lorawan-device/Cargo.toml

```diff
@@ -19,17 +19,17 @@ rustdoc-args = ["--cfg", "docsrs"]
 
 [dependencies]
 lora-modulation = { version = ">=0.1.2", default-features = false }
-lorawan = { version = "0.7.4", path = "../lorawan-encoding", default-features = false }
+lorawan = { path = "../lorawan-encoding", default-features = false }
 heapless = "0.7"
-generic-array = "0.14.2"
+generic-array = "0.14"
 defmt = { version = "0.3", optional = true }
 fastrand = { version = "2", default-features = false }
 futures = { version = "0.3", default-features = false, optional = true }
 rand_core = { version = "0.6", default-features = false }
 serde = { version = "1", default-features = false, features = ["derive"], optional = true }
 seq-macro = "0.3.5"
 document-features = "0.2.7"
-embassy-time = { version = "0.1.5", optional = true }
+embassy-time = { git = "https://github.com/embassy-rs/embassy", rev = "0027a76bb61f19fcae1fe588ba3ff62660d7f7e3", optional = true }
```

Comment:
don't use git in `[dependencies]`, it makes it too easy to end up with multiple incompatible `embassy-time` versions which then fails to link. 

This is why the previous code used `crates.io` versions and then replaced them all to git with a single `[patch.crates-io]`. This ensures all `embassy-time` crates in the tree use the same git rev, so no incompatibiltiies.